### PR TITLE
Move download improvements from Internal to everyone

### PR DESCRIPTION
**Pull request by anikiki**
Task/Issue URL: https://app.asana.com/0/0/1201717030838678/f

### Description
Made the new approach for downloads available to everyone.

### Steps to test this PR

- fresh install from this branch using the Play flavor
- download this image: https://miro.medium.com/max/1050/0*BRkMUINdhCYPk1KM.jpg
- notice it works as expected
- download this pdf: https://www.jdwetherspoon.com/~/media/files/pdf-documents/menus/currentmenus/menu_2103.pdf
- notice it works as expected

### CODE REVIEW

File: app/src/main/java/com/duckduckgo/app/browser/downloader/FilenameExtractor.kt [hunk 1]
```diff
@@ -74,10 +70,7 @@ class FilenameExtractor @Inject constructor(
         mimeType: String?
     ): String {
         val tidiedUrl = url.removeSuffix("/")
-        var guessedFilename = when (appBuildConfig.flavor) {
-            BuildFlavor.INTERNAL -> DownloaderUtil.guessFileName(tidiedUrl, contentDisposition, mimeType)
-            else -> URLUtil.guessFileName(tidiedUrl, contentDisposition, mimeType)
-        }
+        var guessedFilename = DownloaderUtil.guessFileName(tidiedUrl, contentDisposition, mimeType)
```
Comment (anikiki) [inline app/src/main/java/com/duckduckgo/app/browser/downloader/FilenameExtractor.kt:73]: This used to be available only for internal builds, now it's available on all builds.

File: app/src/androidTest/java/com/duckduckgo/app/browser/downloader/UriUtilsFilenameExtractorInternalTest.kt [hunk 2]
```diff
@@ -1,326 +0,0 @@
...
-import com.duckduckgo.appbuildconfig.api.BuildFlavor
-import org.mockito.kotlin.mock
-import org.mockito.kotlin.verify
-import org.junit.Assert.assertEquals
-import org.junit.Assert.assertTrue
-import org.junit.Before
-import org.junit.Test
-import org.mockito.kotlin.whenever
-
-class UriUtilsFilenameExtractorInternalTest {
```
Comment (anikiki) [inline app/src/androidTest/java/com/duckduckgo/app/browser/downloader/UriUtilsFilenameExtractorInternalTest.kt:32]: Deleted as it's not needed anymore. We used to have 2 identical test files, one used to test the implementation for downloaded file name available only for internal builds and the other one for the rest of the builds. Now we have one implementation everywhere and will keep only one set of tests.

File: app/src/main/java/com/duckduckgo/app/browser/downloader/NetworkFileDownloader.kt [hunk 3]
```diff
@@ -34,79 +32,10 @@ import javax.inject.Inject
...
-            ) {
-                // TODO [Improve downloads] This is a connection failed, the reason provided is misleading.
-                callback.downloadFailed(context.getString(R.string.downloadManagerDisabled), DownloadFailReason.DownloadManagerDisabled)
-                return
-            }
-        })
-    }
-
-    private fun downloadV2(pendingDownload: PendingFileDownload, callback: FileDownloader.FileDownloadListener) {
+    fun download(pendingDownload: PendingFileDownload, callback: FileDownloader.FileDownloadListener) {
```
Comment (anikiki) [inline app/src/main/java/com/duckduckgo/app/browser/downloader/NetworkFileDownloader.kt:38]: This used to be the implementation available for internal builds only and it's now the one to be used everywhere.

**Review by malmstein — APPROVED** [review]
Nice work @anikiki !

**Pull request merged**